--- a/Modules/timemodule.c
+++ b/Modules/timemodule.c
@@ -36,6 +36,14 @@
 
 #define SEC_TO_NS (1000 * 1000 * 1000)
 
+#include "emscripten.h"
+EM_JS(int, PYJS_Sleep, (double t), {
+    if (typeof JS_Sleep === 'function') {
+        return JS_Sleep(t);
+    }
+    return 0;
+});
+
 /* Forward declarations */
 static int pysleep(_PyTime_t);
 
@@ -1740,6 +1747,9 @@
 static int
 pysleep(_PyTime_t secs)
 {
+#ifdef __EMSCRIPTEN__
+    return PYJS_Sleep(_PyTime_AsSecondsDouble(secs));
+#else
     _PyTime_t deadline, monotonic;
 #ifndef MS_WINDOWS
     struct timeval timeout;
@@ -1811,4 +1821,5 @@
     } while (1);
 
     return 0;
+#endif
 }
