diff --git a/Python/ceval.c b/Python/ceval.c
index df5c093..8c95c0c 100644
--- a/Python/ceval.c
+++ b/Python/ceval.c
@@ -1,3 +1,13 @@
+#include "emscripten.h"
+
+EM_JS(int, PYJS_KeyboardInterrupt, (void), {
+    if (typeof JS_KeyboardInterrupt === 'function') {
+        if (JS_KeyboardInterrupt()) {
+            return 1;
+        }
+    }
+    return 0;
+});
 
 /* Execute compiled code */
 
@@ -940,6 +950,11 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
            async I/O handler); see Py_AddPendingCall() and
            Py_MakePendingCalls() above. */
 
+        if (PYJS_KeyboardInterrupt())
+        {
+            PyErr_SetString(PyExc_KeyboardInterrupt, "");
+            goto error;
+        }
         if (_Py_atomic_load_relaxed(&_PyRuntime.ceval.eval_breaker)) {
             if (_Py_OPCODE(*next_instr) == SETUP_FINALLY ||
                 _Py_OPCODE(*next_instr) == YIELD_FROM) {
