diff --git a/Python/ceval.c b/Python/ceval.c
index 3d69038..2a78f2b 100644
--- a/Python/ceval.c
+++ b/Python/ceval.c
@@ -621,6 +621,7 @@ Py_MakePendingCalls(void)
 */
 
 #define NPENDINGCALLS 32
+#define NPENDINGCALLS_HANDLE 1
 static struct {
     int (*func)(void *);
     void *arg;
@@ -662,7 +663,8 @@ Py_MakePendingCalls(void)
         return 0;
     busy = 1;
     UNSIGNAL_PENDING_CALLS();
-    for (;;) {
+    int c;
+    for (c = 0; c < NPENDINGCALLS_HANDLE; c++) {
         int i;
         int (*func)(void *);
         void *arg;
